Makefile

```diff
@@ -11,11 +11,13 @@ GCC = g++
 endif
 LD = $(GCC)
 
-ifndef RELEASE
-CFLAGS += -Werror -Wimplicit-fallthrough
+ifdef LLD
```

Comment:
Something is wrong when compiling with latest clang (6.0) and linking with lld. I get following errors:
> defined in /usr/lib/x86_64-linux-gnu/libopenal.so
> referenced by audio_device.cpp:17
>               obj-opt/audio_device.o:(_ZN11AudioDevice10initializeB5cxx11Ev) /usr/lib/llvm-6.0/bin/ld.lld: error: cannot preempt symbol: alcCreateContext
> defined in /usr/lib/x86_64-linux-gnu/libopenal.so
> referenced by audio_device.cpp:18
>               obj-opt/audio_device.o:(_ZN11AudioDevice10initializeB5cxx11Ev)
...

Reply:
I tried out clang 6.0 and lld works fine for me, so I think it's something specific to your system, maybe the version of openal. In any case I think it's more convenient to use lld by default and add a flag like NO_LLD.

Reply:
OK, I'll see what's going on here and if it still won't work, I'll change it to NO_LLD.